Update docs (#2)
cycloid-community-catalog/stack-concourse-server: README.md

Concourse is an open-source continuous thing-doer.

> Built on the simple mechanics of resources, tasks, and jobs, Concourse presents a general approach to automation that makes it great for CI/CD.

Reviewer: ```suggestion
Built on the simple mechanics of resources, tasks, and jobs. Concourse presents a general approach to automation that makes it great for CI/CD.
```

# Architecture

This stack will deploy a Concourse server (web+atc) on an Amazon EC2 instance behind ALB/NLB loadbalancers, using PostgreSQL for Cycloid.io.

<p align="center">
<img src="docs/diagram.png" width="400">
</p>

  * **ALB**: Amazon application loadbalancer
  * **NLB**: Amazon network loadbalancer
  * **ASG**: Autoscaling group for fronts
  * **concourse-server**: EC2 instances from builded AMI
  * **RDS**: Amazon RDS database (PostgreSQL)

# Requirements

In order to run this task, couple elements are required within the infrastructure:

>   * Having a VPC with private & public subnets [Here](https://docs.aws.amazon.com/vpc/latest/userguide/getting-started-ipv4.html#getting-started-create-vpc)

Reviewer: ```suggestion
  * Having a VPC with private & public subnets [here](https://docs.aws.amazon.com/vpc/latest/userguide/getting-started-ipv4.html#getting-started-create-vpc)
```

>   * Having an S3 bucket to store Terraform remote states [Here](https://docs.aws.amazon.com/quickstarts/latest/s3backup/step-1-create-bucket.html)

Reviewer: ```suggestion
  * Having an S3 bucket to store Terraform remote states [here](https://docs.aws.amazon.com/quickstarts/latest/s3backup/step-1-create-bucket.html)
```

# Details

## Pipeline

> **Note** The pipeline contains a manual approval between terraform plan and terraform apply.
> That means if you trigger a terraform plan, to apply it, you have to go on terraform apply job
> and click on the `+` button to trigger it.

<img src="docs/pipeline.png" width="800">

**Jobs description**

  * `terraform-plan`: Terraform job that will simply make a plan of the stack.
  * `terraform-apply`: Terraform job similar to the plan one, but will actually create/update everything that needs to. Please see the plan diff for a better understanding.
  * `terraform-destroy`: :warning: Terraform job meant to destroy the whole stack - **NO CONFIRMATION ASKED**. If triggered, the full project **WILL** be destroyed. Use with caution.
  * `build-ami` : Build Concourse server Amazon image AMI using Packer and Ansible.
  * `clean-ami` : Remove old AMI of Cycloid workers.

> 

Reviewer: I'd reorder by jobs order.
The deploy AMI isn't present here, is it still required after an apply, or forgotten?

**Params**

|Name|Description|Type|Default|Required|
|---|---|:---:|:---:|:---:|

> |`aws_access_key`|Amazon AWS access key for Terraform. see value format [Here](https://docs.cycloid.io/advanced-guide/integrate-and-use-cycloid-credentials-manager.html#vault-in-the-pipeline)|`-`|`((aws.access_key))`|`True`|

Reviewer: ```suggestion
|`aws_access_key`|Amazon AWS access key for Terraform. see value format [here](https://docs.cycloid.io/advanced-guide/integrate-and-use-cycloid-credentials-manager.html#vault-in-the-pipeline)|`-`|`((aws.access_key))`|`True`|
```

|`aws_default_region`|Amazon AWS region to use for Terraform.|`-`|`eu-west-1`|`True`|

> |`aws_secret_key`|Amazon AWS secret key for Terraform. see value format [Here](https://docs.cycloid.io/advanced-guide/integrate-and-use-cycloid-credentials-manager.html#vault-in-the-pipeline)|`-`|`((aws.secret_key))`|`True`|

Reviewer: ```suggestion
|`aws_secret_key`|Amazon AWS secret key for Terraform. see value format [here](https://docs.cycloid.io/advanced-guide/integrate-and-use-cycloid-credentials-manager.html#vault-in-the-pipeline)|`-`|`((aws.secret_key))`|`True`|
```

|`rds_type`|RDS database instance size.|`-`|`db.t3.small`|`False`|
|`rds_username`|RDS database username.|`-`|`concourse`|`False`|
|`vpc_id`|Amazon VPC id on which create each components.|`-`|``|`True`|
|`workers_cidr_allow`|List of CIDR of Concourse workers to allow.|`list`|``|`True`|
|`workers_sg_allow`|List of workers security groups of Concourse workers to allow.|`list`|``|`True`|

**Outputs**

| Name | Description |
|------|-------------|
| alb_dns_name | DNS name of the ALB on top of Concourse servers |
| alb_zone_id | Zone ID of the ALB on top of Concourse servers |
| asg_concourse_sec_group_id | Security group ID of Concourse servers |
| nlb_dns_name | DNS name of the NLB on top of Concourse servers |
| nlb_zone_id | Zone ID of the NLB on top of Concourse servers |
| rds_address | Address of the RDS database. |
| rds_database | Database name of the RDS database. |
| rds_port | Port of the RDS database. |
| rds_username | Username of the RDS database. |

## Ansible

  * Playbook and packer config to build a debian image with telegraf, fluentd and Concourse server (web+atc) installed.

> > Note: To allow concourse external workers, you will need to their SSH public keys

Reviewer: ```suggestion
> Note: To allow concourse external workers, you will need to put their SSH public key
```

> in your ansible stack config in the following directory:
> * `{PROJECT_NAME}/ansible/files/worker_keys/`

> > Allowed worker keys in this directory will be assembled into a authorized_keys

Reviewer: ```suggestion
> Allowed worker keys in this directory will be assembled into an authorized_keys
```
